features/config/dao/impl/src/main/java/org/opennms/features/config/dao/impl/util/OpenAPIBuilder.java

```diff
@@ -0,0 +1,159 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019-2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ ******************************************************************************/
+package org.opennms.features.config.dao.impl.util;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import org.opennms.features.config.dao.api.ConfigItem;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class OpenAPIBuilder {
```

Comment:
@freddy61025,
how to I get from an existing OpenAPI Object to a ConfigItem?
I think we need a way to modify existing OpenAPI definitions.
I would need something like OpenAPIBuilder.createBuilderFromExisting(OpenAPI openAPI) in order to modify the Properties schema. Is that possible?